2009-05-16  Aurelien Jarno  <aurelien@aurel32.net>

        * sysdeps/unix/sysv/linux/alpha/getsysstats.c (GET_NPROCS_PARSER):
        Change parameters and use next_line.

---
 ports/sysdeps/unix/sysv/linux/alpha/getsysstats.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- ports/sysdeps/unix/sysv/linux/alpha/getsysstats.c
+++ ports/sysdeps/unix/sysv/linux/alpha/getsysstats.c
@@ -20,15 +20,16 @@
 
 
 /* We need to define a special parser for /proc/cpuinfo.  */
-#define GET_NPROCS_PARSER(FP, BUFFER, RESULT)				   \
+#define GET_NPROCS_PARSER(FD, BUFFER, CP, RE, BUFFER_END, RESULT)	   \
   do									   \
     {									   \
       /* Find the line that contains the information about the number of   \
 	 active cpus.  We don't have to fear extremely long lines since	   \
 	 the kernel will not generate them.  8192 bytes are really enough. \
 	 If there is no "CPUs ..." line then we are on a UP system.  */	   \
+      char *l;								   \
       (RESULT) = 1;							   \
-      while (fgets_unlocked (BUFFER, sizeof (BUFFER), FP) != NULL)	   \
+      while ((l = next_line (FD, BUFFER, &CP, &RE, BUFFER_END)) != NULL)  \
 	if ((sscanf (BUFFER, "cpus active : %d", &(RESULT)) == 1)	   \
 	    || (sscanf (BUFFER, "CPUs probed %*d active %d",		   \
 			&(RESULT)) == 1))  				   \
